src/ccl_halomod.c

```diff
@@ -0,0 +1,428 @@
+#include "ccl_utils.h"
+#include "math.h"
+#include "stdio.h"
+#include "stdlib.h"
+#include "gsl/gsl_integration.h"
+#include "gsl/gsl_interp.h"
+#include "gsl/gsl_spline.h"
+#include "gsl/gsl_sf_expint.h"
+#include "ccl_placeholder.h"
+#include "ccl_background.h"
+#include "ccl_power.h"
+#include "ccl_massfunc.h"
+#include "ccl_error.h"
+#include "class.h"
+#include "ccl_params.h"
+#include "ccl_emu17.h"
+#include "ccl_emu17_params.h"
+
+// Set two-halo term
+// 1 - Standard two-halo term
+// 2 - Linear theory
+int i2h=1;
+
+// Set full halo model pwoer spectrum method
+// 1 - Standard sum of two- and one-halo terms
+// 2 - Smooth transition using alpha parameter from Mead et al. (2015)
+int ipow=1;
+
+// Set concentration-mass relation
+// 1 - Bhattaharya et al. (2011)
+// 2 - Full Bullock et al. (2001)
+// 3 - Duffy et al. (2008)
+// 4 - Constant concentration (for testing)
+// 5 - Simple Bullock et al. (2001)
+int iconc=3;
+
+// Select window function
+// 1 - NFW profile, appropriate for matter power spectrum
+int iwin=1;
+
+// Set the mass range for the halo-model integration
+double mmin=1e7;
+double mmax=1e17;
+
+// Virial density of haloes
+double Delta_v=200.;
+
+// HMcode parameter: one-halo damping wavenumber: k*  
+double k_star(ccl_cosmology *cosmo, double a, int * status){  
+  return 0.584/ccl_sigmaVz(cosmo, 0., a, status);
+}
+
+// HMcode parameter: two-halo damping: f  
+double f_damp(ccl_cosmology *cosmo, double a, int * status){ 
+  // Damping f from Mead et al. (2015)
+  //double sigma8z = ccl_sigmaRz(cosmo, 8., a, status);
+  //return 0.188*pow(sigma8z, 4.29);
+
+  //Damping f from Mead et al. (2016)
+  double sigmav100 = ccl_sigmaVz(cosmo, 100., a, status);
+  return 0.0095*pow(sigmav100, 1.37); 
+}
+
+// HMcode parameter: spectral index at the collapse scale: neff
+// TODO: Actually code this up
+double collapse_index(ccl_cosmology *cosmo, double a, int * status){
+  return -2.;
+}
+
+// HMcode parameter: smoothing for transition between one- and two-halo regimes: alpha
+double alpha(ccl_cosmology *cosmo, double a, int * status){  
+  double neff = collapse_index(cosmo, a, status);
+  //return 2.93*pow(1.77,neff); //This is the relation from Mead et al. (2015)
+  return 3.24*pow(1.85,neff); //This is the relation from Mead et al. (2016)
+}
```

Comment:
Same for all of these functions

Reply:
Okay, these functions I added in because they pertain to the HMcode that I plan to add at a later date. So they will eventually be used. What is the protocol here? Should I make a new branch for HMcode and just leave all this half-finished stuff there? Or is it okay to leave the half-finished functions here for the time being?

Reply:
Done